### PR TITLE
feat: vitest serializer extract path utils

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Description

* Extract path utils from `@scripts/vitest-config` to `@modern-js/utils`.
* Simplify option definitions of `createSnapshotSerializer`.

> All snapshot changes have been checked.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/toolkit/utils/src/path.ts [hunk 1]
```diff
@@ -22,3 +23,50 @@ export const getTemplatePath = (prefix?: string) => {
...
+  let ret: string | null = null;
+  try {
+    ret = os.tmpdir();
+    ret = fs.realpathSync(ret);
+  } catch {}
+  return ret;
+}
+
+export function splitPathString(str: string) {
+  return str.split(/[\\/]/);
```
Comment (chenjiahan) [inline packages/toolkit/utils/src/path.ts:71]: how about `str.split(path.seq)`

File: packages/toolkit/utils/src/path.ts [hunk 2]
```diff
@@ -22,3 +23,50 @@ export const getTemplatePath = (prefix?: string) => {
...
+  return _(start)
+    .split(/[/\\]/)
+    .filter(Boolean)
+    .map(_joinPathParts)
+    .reverse()
+    .push('/')
+    .value();
+}
+
+export function realTemplateDirectory() {
```
Comment (chenjiahan) [inline packages/toolkit/utils/src/path.ts:61]: Template -> Temporary ?

File: packages/toolkit/utils/src/path.ts (hunk 1 above)
Comment (Asuka109) [reply packages/toolkit/utils/src/path.ts:71]: 现在是有 posix 和 win32 路径的测试用例交叉运行的 case，以及需要在 win32 平台上同时处理两种风格的路径，所以做成这样了

File: packages/toolkit/utils/src/pathSerializer.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import os from 'os';
```
[the remark this answers is not in the document]
Comment (Asuka109) [reply packages/toolkit/utils/src/pathSerializer.ts:1]: friendly-errors 插件里后面应该会用到的

File: packages/toolkit/utils/src/path.ts (hunk 2 above)
Comment (Asuka109) [reply packages/toolkit/utils/src/path.ts]: done.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**